### DISCUSSION TITLE
Controlling the motor controller from the Arduino

### DISCUSSION THREAD

**Discussion by Saverio911** [AlexAsplund/Vanchor#21] [Q&A]
This is an amazing idea. I am giving it a go. I am trying to figure out how you use the Arduino to control the motor speed with PWM. Are you intercepting the signal from a 555 timer? Is there a wiring diagram that shows where the Arduino is connected to the motor controller? Great project.

**Comment by AlexAsplund** [comment]
On the first 2 controllers (that I broke) I just sent PWM to the signal pin where the potentiometer would go. On the latest controller that didn't work for some reason so I just traced the output from the microcontroller on the board to pin 3 on a mosfet driver (or buck converter, didn't find a datasheet).

The walfront boards seems to vary a lot tho! So be sure to measure and verify that it's the correct pin. You should see a change in voltage on the driver pin when you adjust the potentiometer.


Made this masterpiece on my phone since I'm on the go, but the one in the yellow circle is the driver and the crossed over one is the microcontroller I desoldered:

![AC261F86-6D4C-451A-BB7C-10F966873730](https://user-images.githubusercontent.com/12714125/136671253-088b6a89-3014-4d19-bf86-6706cc07acdb.png)

**Reply by Saverio911** [reply]
I thought that might have been the method after I saw the way they used a 555 timer like in this video: https://www.youtube.com/watch?v=ipFxYQkB5uw

**Reply by AlexAsplund** [reply]
I saw that one too. But it isn't applicable on the boards I've had since it's the microcontroller that generates the pwm instead of an 555.

And thanks btw! Was in a hurry yesterday so forgot to write that😅


And here's a bonus picture of the box with all the electronics (except the compass). At first it was tidy but after switching out the controller 2 times and a few other things I just decided to wait with organizing it 😁


![image](https://user-images.githubusercontent.com/12714125/136689923-4eb11fbc-41fc-464e-bb4e-d0871351489e.jpeg)
